examples/04_tool_inference_lock/ex_tool_inference_lock/vendor/inference_lock.rb

```diff
@@ -0,0 +1,147 @@
+# @see https://github.com/Eneroth3/inference-lock
+
+# Mixin for inference lock in tools.
+#
+# If any of the following Tool API methods are defined in the Tool class,
+# they must call `super` for this mixin to function:
+#
+# * `activate`
+# * `onKeyDown`
+# * `onKeyUp`
+# * `onKeyUp`
+# * `onMouseMove`
+#
+# `current_ip` must be implemented by the Tool class for this mixin to function.
+#
+# `start_ip` may be implemented for finer control.
+module Examples::ToolInferenceLock::InferenceLock
+  # @api
+  # @see https://ruby.sketchup.com/Sketchup/Tool.html
+  def activate
+    super if defined?(super)
+
+    @axis_lock = nil
+    @mouse_x = nil
+    @mouse_y = nil
+  end
+
+  # @api
+  # @see https://ruby.sketchup.com/Sketchup/Tool.html
+  def onKeyDown(key, _repeat, _flags, view)
+    if key == CONSTRAIN_MODIFIER_KEY
+      try_lock_constraint(view)
+    else
+      try_lock_axis(key, view)
+    end
+
+    # Emulate mouse move to update InputPoint picking.
+    onMouseMove(0, @mouse_x, @mouse_y, view) if @mouse_x && @mouse_y
+    view.invalidate
+  end
+
+  # @api
+  # @see https://ruby.sketchup.com/Sketchup/Tool.html
+  def onKeyUp(key, _repeat, _flags, view)
+    return unless key == CONSTRAIN_MODIFIER_KEY
+    return if @axis_lock
+
+    # Unlock inference.
```

Comment:
This comment as it stands is confusing, since it contradicts the name of the method underneath. I think it should elaborate that calling it without any arguments unlocks it.